File: fsharp/FsSearch/FsSearchTests/EmbeddedTestResource.fs
namespace FsSearchTests

open System.IO
open System.Reflection

module EmbeddedTestResource = 

    let GetResourceFileContents (namespaceAndFileName : string) : string = 
        let contents =
            try
                use stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(namespaceAndFileName)
                use reader = new StreamReader(stream)
                reader.ReadToEnd()
            with
            | :? IOException as e -> printfn $"%s{e.Message}"; ""
        contents

File: csharp/CsSearch/CsSearchTests/EmbeddedTestResource.cs
using System;
using System.IO;
using System.Reflection;
using System.Text;

namespace CsSearchTests
{
    public static class EmbeddedTestResource
    {
        public static string GetResourceFileContents(string namespaceAndFileName)
        {
            try
            {
                using var stream = typeof(EmbeddedTestResource).GetTypeInfo().Assembly
                    .GetManifestResourceStream(namespaceAndFileName);
                using var reader = new StreamReader(stream!, Encoding.UTF8);
                return reader.ReadToEnd();
            }
            catch(Exception)
            {
                throw new Exception($"Failed to read Embedded Resource {namespaceAndFileName}");
            }
        }
    }
}
